### PR TITLE
Server Config (#734, #682)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #734, #682

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1704,6 +1702,7 @@ def __init_server(self):
                 path_mapping=self._path_mapping,
                 flask=self._flask,
                 async_mode=app_config["async_mode"],
+                server_config=app_config["server_config"],
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py]: should we use app_config.get("server_config") here ?

File: src/taipy/gui/gui.py [hunk 2]
```diff
@@ -1715,6 +1714,7 @@ def __init_server(self):
...
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py]: and here

**Review by FredLL-Avaiga — COMMENTED** [review]
Looks like szerver_config can be None ?

File: src/taipy/gui/server.py [hunk 3]
```diff
@@ -50,26 +51,45 @@ def __init__(
...
         gui: Gui,
         flask: t.Optional[Flask] = None,
-        path_mapping: t.Optional[dict] = {},
+        path_mapping: t.Optional[dict] = None,
         async_mode: t.Optional[str] = None,
+        server_config: t.Optional[ServerConfig] = None,
     ):
         self._gui = gui
+        server_config = server_config or {}
         self._flask = Flask("Taipy") if flask is None else flask
```
Comment (FabienLelaquais) [inline src/taipy/gui/server.py]: Do we want to let users customize the name of the application package here?
Adding a slot in ServerConfig..?

**Review by FabienLelaquais — APPROVED** [review]
Yes I like that a lot!
Well done!

File: src/taipy/gui/server.py (hunk 3 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/server.py]: I have made it so you configure all flask init parameter

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**